examples/blas/compile_time_dispatching/level3/gemm_usm_mklgpu.cpp

```diff
@@ -0,0 +1,211 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+/*
+*
+*  Content:
+*       This example demonstrates use of DPCPP API oneapi::mkl::blas::gemm
+*       using unified shared memory to perform General
+*       Matrix-Matrix Multiplication on an Intel GPU SYCL device.
+*
+*       C = alpha * op(A) * op(B) + beta * C
+*
+*       where op() is defined by one of oneapi::mkl::transpose::{nontrans,trans,conjtrans}
+*
+*
+*       The supported floating point data types for gemm matrix data are:
+*           float
+*
+*
+*******************************************************************************/
+
+
+// stl includes
+#include <iostream>
+#include <cstdlib>
+#include <limits>
+#include <vector>
+#include <algorithm>
+#include <cstring>
+#include <list>
+#include <iterator>
+
+
+#include <CL/sycl.hpp>
+#include "oneapi/mkl.hpp"
+
+#include "example_helper.hpp"
+
+//
+// Main example for Gemm consisting of
+// initialization of A, B and C matrices as well as
+// scalars alpha and beta.  Then the product
+//
+// C = alpha * op(A) * op(B) + beta * C
+//
+// is performed and finally the results are post processed.
+//
+void run_gemm_example(const sycl::device &dev) {
+
+    //
+    // Initialize data for Gemm
+    //
+    // C = alpha * op(A) * op(B)  + beta * C
+    //
+
+    oneapi::mkl::transpose transA = oneapi::mkl::transpose::trans;
+    oneapi::mkl::transpose transB = oneapi::mkl::transpose::nontrans;
+
+    // matrix data sizes
+    int m = 45;
+    int n = 98;
+    int k = 67;
+
+    // leading dimensions of data
+    int ldA = 103;
+    int ldB = 105;
+    int ldC = 106;
+    int sizea, sizeb, sizec = ldC * n;
+
+    // set scalar fp values
+    float alpha = set_fp_value(float(2.0), float(-0.5));
+    float beta  = set_fp_value(float(3.0), float(-1.5));
+
+    // Catch asynchronous exceptions
+    auto exception_handler = [] (sycl::exception_list exceptions) {
+        for (std::exception_ptr const& e : exceptions) {
+            try {
+                std::rethrow_exception(e);
+            } catch(sycl::exception const& e) {
+                std::cout << "Caught asynchronous SYCL exception during GEMM:\n"
+                << e.what() << std::endl;
+            }
+        }
+    };
```

Comment:
this lambda should exit with some error code when loop finishes, perhaps.
```std::exit(2)``` or something

Reply:
Addressed in latest commits